### ISSUE TITLE
Support schema initialization for Microsoft SQL Server

### ISSUE THREAD

**Issue by davidkarlsen** [spring-projects/spring-modulith#804]
SQL-server isn't a supported database type as of today:

```
Caused by: java.lang.IllegalArgumentException: Unsupported database type: Microsoft SQL Server
	at org.springframework.modulith.events.jdbc.DatabaseType.lambda$from$1(DatabaseType.java:54)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at org.springframework.modulith.events.jdbc.DatabaseType.from(DatabaseType.java:54)
	at org.springframework.modulith.events.jdbc.JdbcEventPublicationAutoConfiguration.databaseType(JdbcEventPublicationAutoConfiguration.java:45)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:146)
	... 42 more
```

implement support for this.

**Comment by odrotbohm** [comment]
That's a reasonable suggestion. I've tried to get a Testcontainers setup working for an hour now and failed to get a MSSQL server instance boot on my M1 Mac properly via Docker. If anyone was willing to contribute such a setup (create a `application-mssql.properties` with the necessary and augment, for example, `DatabaseSchemaInitializerIntegrationTests` with an MSSQL-specific execution) I am happy to take another look.

**Comment by davidkarlsen** [comment]
We'll get back with a PR (and perhaps questions)

**Comment by davidkarlsen** [comment]
@odrotbohm PTAL

**Comment by odrotbohm** [comment]
Nope. As indicated in the fix version, I've merged this for 1.3 only as we generally only add new features in minor versions for stability reasons. We're going to ship the RC next week, i.e. it's close to being available in GA form.

Any particular reason you hoped to see it in 1.2?
